### ISSUE TITLE
Bug: [consistent-type-imports] imported type used in 'satisfies' reported as 'used by decorator metadata'

### ISSUE THREAD

**Issue by jet2jet** [typescript-eslint/typescript-eslint#6077]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play#ts=4.9.3&sourceType=module&code=PTAEAEBcGcFoFMAeAHeBjSCBOWD2WBYAKAEsBbZfSUSAT1VAG9QAxXXUAX1ADM8zQAch7tBAbmJpcAO2jURHALxNioUAEMAXEIWDi3aOsgloPEvGit2EokkpZqAE3g91AVwA2868SA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMkRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
// @ts-expect-error
import type { Foo } from 'foo';
const foo = {
  a: 'foo'
} satisfies Foo;
export default foo;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-imports": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error reported

### Actual Result

Error reported: `Type import "Foo" is used by decorator metadata. 2:1 - 2:32`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.44.0` |
| `@typescript-eslint/parser`            | `5.44.0` |
| `TypeScript`                           | `4.9.3` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `18.12.0` |

**Comment by bradzacher** [comment]
This was fixed in #6059 and will be released tomorrow.
I.e. this is a duplicate of #6057

**Comment by kasperpeulen** [comment]
@bradzacher 

I tried this out in the latest canary, but it seems to still cause issues:
```ts
import type { Meta, StoryObj } from '@storybook/react';

import { Button } from './Button';

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta = {
  title: 'Example/Button',
  component: Button,
  tags: ['docsPage'],
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/7.0/react/writing-stories/args
export const Primary: Story = {
  args: {
    primary: true,
    label: 'Button',
  },
};
```

```
/Users/kasperpeulen/code/storybook/storybook/code/renderers/react/template/cli/ts/Button.stories.ts
  1:1  error  Type import "Meta" is used by decorator metadata  @typescript-eslint/consistent-type-imports
```


Using `5.44.1-alpha.15`

**Comment by kasperpeulen** [comment]
@bradzacher 

I tried this out in the latest canary, but it seems to still cause issues:
```ts
import type { Meta, StoryObj } from '@storybook/react';

import { Button } from './Button';

// More on how to set up stories at: https://storybook.js.org/docs/7.0/react/writing-stories/introduction
const meta = {
  title: 'Example/Button',
  component: Button,
  tags: ['docsPage'],
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} satisfies Meta<typeof Button>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/7.0/react/writing-stories/args
export const Primary: Story = {
  args: {
    primary: true,
    label: 'Button',
  },
};
```

```
/Users/kasperpeulen/code/storybook/storybook/code/renderers/react/template/cli/ts/Button.stories.ts
  1:1  error  Type import "Meta" is used by decorator metadata  @typescript-eslint/consistent-type-imports
```


Using `5.44.1-alpha.15`

**Comment by jet2jet** [comment]
It seems this issue was fixed in version `5.45.0` (at least no error is reported in `5.46.0`).
I'll close this issue.
